getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,49 +1463,88 @@ freed.
 
-Signals are a way to send notification messages from an object that
-other objects can listen to in a generic way. Create custom signals for
-a class using the ``signal`` keyword.
+Signals are a tool to emit messages from an object that other objects can react
+to. To create custom signals for a class, use the ``signal`` keyword.
 
 ::
 
-   # Signal with no arguments
-   signal your_signal_name
+   # A signal named health_depleted
+   signal health_depleted
 
-   # Signal with two arguments
-   signal your_signal_name_with_args(a, b)
+   # A signal that forwards two arguments
+   signal health_changed(old_value, new_value)
 
-These signals may be connected to methods in the same manner as you connect
-built-in signals of nodes such as :ref:`class_Button` or :ref:`class_RigidBody`.
+.. note:: 
 
-Here's an example that creates a custom signal in one script and connects
-the custom signal to a method in a separate script, using the
+   Signals are a `Callback
+          <https://www.wikiwand.com/en/Callback_(computer_programming)>`
+          mechanism. They also fill the role of Observers, a common programming
+          pattern. For more information, read the `Observer
+          tutorial<http://gameprogrammingpatterns.com/observer.html>`_ in the
+          Game Programming Patterns ebook.
+
+You can connect these signals to methods the same way you connect built-in
+signals of nodes like :ref:`class_Button` or :ref:`class_RigidBody`.
+
+Here's an example of when signals can be useful. 
+
+Let's say we have a game where the playable character can take damage from
+enemies. We want a life bar on screen to react to health changes with an
+animation, but we want to keep the User Interface separate from the player in
+our scene tree.
+
+In our ``Character.gd`` script, we define a ``health_changed`` signal and emit
+it with :ref:`Object.emit_signal() <class_Object_method_emit_signal>`, and from
+a ``Game`` node higher up our scene tree, we connect it to the ``Lifebar`` using the
 :ref:`Object.connect() <class_Object_method_connect>` method:
 
 ::
 
-   # your_notifier.gd
+    # Character.gd
 
-   signal data_found
+    ...
+    signal health_changed(old_value, new_value)
 
-   var your_data = 42
+    func take_damage(amount):
+        var old_health = health
+        health -= amount
+
+        # We emit the health_changed signal every time the
+        # character took damage
+        emit_signal('health_changed', old_health, health)
+    ...
 
 ::
 
-   # your_handler.gd
+    # Lifebar.gd 
 
-   func your_handler():
-      print("Your handler was called!")
+    # Here, we define a function to use as a callback when the
+    # character's health_changed signal was emitted
+
+    ...
+    func _on_Character_health_changed(old_value, new_value):
+        if old_value > new_value:
+            progress_bar.color = Color.red
+        else:
+            progress_bar.color = Color.green
+        progress_bar.animate(old_value, new_value)
+    ...
```

Comment:
TextureProgess and ProgressBar both don't have a `color` property or an `animate` method. Maybe you could use `modulate` and `value=` instead?

Reply:
Yes, you're right, it'll be simpler

Reply:
If you use `value = ..` it removes the animation part from the example and it'd make it non-functional. 

My goal is to showcase having to meaningful arguments in the emitted signal that are really used by the callback function.

I did not want to use a tween node as in introduces a new object with a more complicated function call.

Reply:
You are already using the values in the `if` though.

Reply:
Yes, so that when the animation starts, the bar's color changes. You can imagine that the animate function would set the bar back to a neutral color after the animation ended.
It's not much but leaving the example the way it is would mean the code would give a weird result on screen: the bar stays red if you lost health, it stays green if you gain health. That's what I meant.